server/api/crud/runtimes/nuclio/function.py

```diff
@@ -293,10 +296,16 @@ def _compile_function_config(
         )
         # since environment variables have a limited size,
         # large serving specs are stored in config maps that are mounted to the pod
+        serving_spec_len = len(serving_spec)
         if (
             can_pass_via_cm
-            and len(serving_spec) >= mlrun.mlconf.httpdb.nuclio.serving_spec_env_cutoff
+            and serving_spec_len >= mlrun.mlconf.httpdb.nuclio.serving_spec_env_cutoff
         ):
```

Comment:
but are we sure that this gives us the number of bytes directly? It is just the len of the object, not the size. It would be more accurate to do `serving_spec.encode('utf-8')`

Reply:
we get `serving_spec` from https://github.com/mlrun/mlrun/blob/e0ca3e679d496ffb4cecf511d0b7a24b9927fe53/mlrun/runtimes/nuclio/serving.py#L666
so I think using len() is sufficient

Reply:
@moranbental i am not sure, could you please share why do you assume that (maybe )?

Reply:
well, since it is a json we format and we dont use symbols or non-utf8 which could increase the underlying size, the usage is most likely be ok. that being said, the correct way is indeed encode and take its length as @rokatyy suggested. `len(serving_spec.encode('utf-8'))` rather than just `len(serving_spec)`